### PR TITLE
refactor(pdp): consume shared utils as published module + service-folder Docker context

### PR DESCRIPTION
**Pull request by mushrafmim**
## Summary

Migrates the Policy Decision Point (PDP) to consume the shared `utils` package as a **published Go module** (`github.com/OpenDIF/opendif-core/exchange/shared/utils`) instead of a local `replace` directive, and updates the PDP Docker build to use the **service directory as its build context**. This removes PDP's build-time dependency on sibling `exchange/shared/*` source being present in the context.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

## Changes Made

- **`go.mod`**: replaced `require github.com/gov-dx-sandbox/exchange/shared/utils v0.0.0` + local `replace ... => ../shared/utils` with a versioned require on `github.com/OpenDIF/opendif-core/exchange/shared/utils` (pseudo-version); added the corresponding `go.sum` entries.
- **`.go` imports** (`main.go`, `internal/config/config.go`, `v1/handler.go`): updated to the new module path.
- **`Dockerfile`**: build context is now the service directory. Copies `go.mod`/`go.sum` and runs `go mod download` (shared module is fetched from the module proxy); removed the `COPY exchange/shared/utils/` step and the repo-root path prefixes.
- **Build contexts updated to match** the new Dockerfile:
  - `exchange/docker-compose.yml` → `context: ./policy-decision-point`
  - `tests/integration/docker-compose.test.yml` → `context: ../../exchange/policy-decision-point`
  - `.github/workflows/policy-decision-point-{publish,validate}.yml` → `context: ${{ env.SERVICE_PATH }}`

## Testing

- [x] I have tested this change locally
- [ ] I have added unit tests for new functionality
- [ ] I have tested edge cases
- [x] All existing tests pass

- `go build -a ./...` and `go vet ./...` pass for PDP.
- `docker build` on `exchange/policy-decision-point` succeeds (exit 0); the compiled binary is present in the final image. The shared module resolves from the public module proxy — no credentials needed in the build (`OpenDIF/opendif-core` is public).

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have checked that there are no merge conflicts

## Additional Notes

- `consent-engine` and `orchestration-engine` still use the repo-root build context + local `replace` for their shared dependencies. This PR intentionally scopes the change to PDP; migrating the other two the same way requires publishing their shared deps (`monitoring`, and OE's root `shared/audit`).
- The `file:` inputs in the workflows remain `${{ env.SERVICE_PATH }}/Dockerfile` — in `docker/build-push-action`, `file` is resolved relative to the workspace, not to `context`.
- There is no `.dockerignore` in the service directory, so build cruft (`coverage.*`, `.env.template`) is sent to the daemon. A small service-level `.dockerignore` could be added as a follow-up.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by sthanikan2000 — DISMISSED** [review]

File: exchange/policy-decision-point/go.mod [hunk 1]
```diff
@@ -3,16 +3,14 @@ module github.com/gov-dx-sandbox/exchange/policy-decision-point
...
+	github.com/OpenDIF/opendif-core/exchange/shared/utils v0.0.0-20260702054005-b41f3c6376d8
 	github.com/google/uuid v1.6.0
-	github.com/gov-dx-sandbox/exchange/shared/utils v0.0.0
 	github.com/stretchr/testify v1.10.0
 	gorm.io/driver/postgres v1.6.0
 	gorm.io/driver/sqlite v1.6.0
 	gorm.io/gorm v1.31.0
 )
 
-replace github.com/gov-dx-sandbox/exchange/shared/utils => ../shared/utils
```
[the remark this answers is not in the document]
Comment (mushrafmim) [reply exchange/policy-decision-point/go.mod:14]: Not Needed.

File: exchange/policy-decision-point/Dockerfile [hunk 2]
```diff
@@ -13,17 +13,15 @@ ARG GIT_COMMIT
...
 # Copy go mod files first for better layer caching
-COPY exchange/policy-decision-point/go.mod exchange/policy-decision-point/go.sum ./exchange/policy-decision-point/
-# Copy shared dependencies
-COPY exchange/shared/utils/ ./exchange/shared/utils/
-
-WORKDIR /app/exchange/policy-decision-point/
+COPY go.mod go.sum ./
+# Shared modules (github.com/OpenDIF/opendif-core/...) are now published and
+# fetched from the module proxy, so no local shared source needs to be copied.
 RUN go mod download
```
[the remark this answers is not in the document]
Comment (mushrafmim) [reply exchange/policy-decision-point/Dockerfile:19]: Not needed for now.

File: exchange/policy-decision-point/Dockerfile [hunk 3]
```diff
@@ -13,17 +13,15 @@ ARG GIT_COMMIT
...
 
-# Copy source code (this layer will be invalidated only on code changes)
-COPY exchange/policy-decision-point/ .
+# Copy source code (this layer is invalidated only on code changes)
+COPY . .
```
[the remark this answers is not in the document]
Comment (mushrafmim) [reply exchange/policy-decision-point/Dockerfile:22]: Added. Thanks!

**Review by sthanikan2000 — APPROVED** [review]
